src/expand.cpp

```diff
@@ -86,6 +87,7 @@ namespace Sass {
 
   Statement* Expand::operator()(Ruleset* r)
   {
+    bool old_at_root_without_rule = this->at_root_without_rule;
```

Comment:
After a second look, shouldn't it be reset on [line 100] [1] too?

[1]: https://github.com/sass/libsass/pull/2089/files#diff-ff25c22e06664808e8ba36fcd39b6439R100